### ISSUE TITLE
Question: Why does the `pkgs_dirs` variable not behave as expected?

### ISSUE THREAD

**Issue by krinsman** [conda/conda#7624]
Here is my `.condarc` file (there are copies both in my home/user directory, as well as in the root installation directory `/opt/miniconda`, and both have the same contents)

```
envs_dirs:
  - /opt/miniconda
pkgs_dirs:
  - /opt/miniconda/pkgs
```

Based on this, I would have expected, when creating a new Conda environment, that the files `urls` and `urls.txt` would be placed in `/opt/miniconda/pkgs`. Instead, they are placed in `/opt/pkgs`.

My expectation is based on my (evidently incorrect) reading of [the relevant conda documentation](https://conda.io/docs/user-guide/configuration/use-condarc.html#specify-package-directories-pkgs-dirs).

I had to manually give my user write permission for `/opt/miniconda` and all objects contained therein. I thought this would have been enough to use Conda based on my `.condarc`, but instead it was also necessary to give my user write permission for `/opt/pkgs` and all objects contained therein.

Is there a way to change the Conda settings yet further so as to make necessary _only_ the directory `/opt/miniconda`? (And make unnecessary using `/opt/pkgs` when creating environments?)

**Comment by krinsman** [comment]
@kalefranz Oops, sorry about forgetting that earlier.

```
    active environment : base
    active env location : /opt/miniconda
            shell level : 1
       user config file : /Users/krinsman/.condarc
 populated config files : /opt/miniconda/.condarc
                          /Users/krinsman/.condarc
          conda version : 4.5.4
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : /opt/miniconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /opt/miniconda/pkgs
       envs directories : /opt/miniconda
                          /opt/miniconda/envs
                          /Users/krinsman/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.4 requests/2.18.4 CPython/3.6.5 Darwin/17.7.0 OSX/10.13.6
                UID:GID : 77075:77075
             netrc file : None
           offline mode : False
```

**Comment by kalefranz** [comment]
I do believe this is a bug.  Can you help me understand the reason you want to set `envs_dirs` to `/opt/miniconda` rather than the standard `/opt/miniconda/envs`?

The root cause of this bug is that an envs_dir and a pkgs_dir both need a "magic file" for conda to use to determine if the directory is writable or read only.  That file for a pkgs_dir is `urls.txt`.  Unfortunately, we haven't yet designated what the magic file should be for envs_dirs.  We started to at one point for 4.4, but then we backed that out.  So we still make the assumption that next to each envs_dir will be a pkgs_dir, and we use the pkgs_dir magic file to test writability for the envs_dir.  With the configuration you're presenting, this isn't a valid assumption.  Indeed, in general, it's not a valid assumption.  It was just a shortcut that hasn't surfaced until now.  I'm glad you filed the issue.

**Comment by krinsman** [comment]
@kalefranz Oh there was no reason for wanting them in `/opt/miniconda` instead of `/opt/miniconda/envs` -- I wanted the environments in the latter, but assumed that the `/envs` part would be added automatically, and that the value of `env_dirs` would be the same as the output of `echo $CONDA_PREFIX` (done from the base environment).

In hindsight I'm not sure why I thought that would work, although it's unclear to me whether any conscious thought had ever went into that decision at all.

So this is probably just me using the software stupidly instead of there actually being a bug -- it wouldn't be the first time.

**Comment by pelson** [comment]
Given the presence of the ``.conda_envs_dir_test`` "magic file" in https://github.com/conda/conda/blob/568a948573a349cfa30b2fbb41ffc56014236887/conda/base/context.py#L1498 I this is now addressed (tested with conda ``4.7.11``). I believe this issue can now be closed.

Looks like it was fixed by #7180

**Comment by krinsman** [comment]
@pelson If you think it's fixed, then feel free to close as resolved. I don't really know either way.
